index.js

```diff
@@ -1460,13 +1460,74 @@ module.exports = function (log, indexesPath) {
     )
   }
 
+  function reindex(offset, cb) {
+    let seq = 0
+    if (offset === 0) {
+      seq = 0
+      offset = -1
+    } else if (offset !== -1) {
```

Comment:
Would be good to have a code comment describing what this block is trying to achieve. So far I'm 2min looking at it trying to understand it.